test/source/TestTypeTraits.cpp

```diff
@@ -1823,6 +1831,20 @@ int TestTypeTraits()
 		#endif
 	}
 
+	// is_final
+	static_assert(std::is_final<FinalStruct>::value, "final struct not correctly detected");
+	static_assert(std::is_final<FinalClass>::value, "final struct not correctly detected");
+	static_assert(!std::is_final<Enum>::value, "enum not correctly detected");
+	static_assert(!std::is_final<int>::value, "int not correctly detected");
+	static_assert(!std::is_final<Struct>::value, "non-final struct not correctly detected");
+	static_assert(!std::is_final<Class>::value, "non-final class not correctly detected");
+	static_assert(is_final<FinalStruct>::value, "final struct not correctly detected");
+	static_assert(is_final<FinalClass>::value, "final struct not correctly detected");
```

Comment:
We should test that our type_trait returns the same answer as the platforms instead of testing them seperately.

Reply:
Sure, makes sense.